src/main/java/org/openstreetmap/atlas/checks/validation/tag/GenericTagCheck.java

```diff
@@ -0,0 +1,940 @@
+        this.popularTagPercentageKey = this
+                .configurationValue(configuration, "tag.percentage_of_key_for_popular",
+                        ((Integer) DEFAULT_TAG_PERCENTAGE_KEY).longValue())
+                .intValue();
+
+        // No matter what the atlas is, we need the db files. But only if we can get
+        // them.
+        if (fileFetcher == null)
+        {
+            return;
+        }
+        for (final String file : Arrays.asList(this.tagInfoDB, this.wikiDataDB))
+        {
+            final Optional<Resource> resource = fileFetcher.apply(file);
+            if (!resource.isPresent() && logger.isDebugEnabled())
+            {
+                logger.debug(MessageFormat.format("{0} is not present", file));
+            }
+        }
+        this.fetchWikiData(fileFetcher);
+        this.fetchTagInfo(fileFetcher);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return !object.getOsmTags().isEmpty()
+                && !this.isFlagged(this.getUniqueOSMIdentifier(object));
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        // Test against each filter and create an instruction if the object passes
+        final Map<String, Collection<IFeatureChange>> instructions = new TreeMap<>();
+
+        if (this.sqliteUtilsTagInfoTagTable != null && this.sqliteUtilsWikiData != null)
+        {
+            if (this.ignoreTags == null || this.ignoreTags.isEmpty())
+            {
+                this.ignoreTags = KeyFullyCheckedUtils.populateIgnoreTags();
+            }
+            this.flagTags(object, instructions);
+        }
+
+        if (!instructions.isEmpty())
+        {
+            // Mark objects flagged by their class and id to allow for the same id in
+            // different object types
+            this.markAsFlagged(this.getUniqueOSMIdentifier(object));
+
+            // Create a flag with generic instructions
+            final String instruction = this.getLocalizedInstruction(
+                    FALLBACK_INSTRUCTIONS.indexOf(INSTRUCTION_INVALID_TAGS),
+                    object.getOsmIdentifier());
+            // If the object is an edge add the edges with the same OSM id
+            final CheckFlag flag = (object instanceof Edge)
+                    ? this.createFlag(((Edge) object).getMainEdge(), instruction)
```

Comment:
The main edge does not encompass all edges with the same ID. It is just the main direction of a way setioned edge. 
 ```suggestion
                    ? this.createFlag(new OsmWayWalker((Edge) object).collectEdges(), instruction)
```